src/libraries/System.Security.Cryptography.X509Certificates/src/Internal/Cryptography/Pal.Unix/CrlCache.cs

```diff
@@ -215,21 +212,15 @@ private static string GetCrlFileName(SafeX509Handle cert, string crlUrl)
             }
 
             uint persistentHash = unchecked((uint)persistentHashLong);
-
-            if (ts_urlHash == null)
-            {
-                ts_urlHash = SHA256.Create();
-            }
-
             Span<byte> hash = stackalloc byte[256 >> 3];
 
             // Endianness isn't important, it just needs to be consistent.
             // (Even if the same storage was used for two different endianness systems it'd stabilize at two files).
             ReadOnlySpan<byte> utf16Url = MemoryMarshal.AsBytes(crlUrl.AsSpan());
 
-            if (!ts_urlHash.TryComputeHash(utf16Url, hash, out int written) || written != hash.Length)
+            if (!SHA256.TryHashData(utf16Url, hash, out int written) || written != hash.Length)
```

Comment:
This could probably be simplified to

```suggestion
            if (SHA256.HashData(utf16Url, hash) != hash.Length)
```